tools/check_spelling.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# Applies requisite code formatters to the source tree
+# check_spelling.sh
+
+set -u
+set -e
+
+VERSION="0.3.4"
```

Comment:
one more thing...from a security perspective can we sha-pin this?

Reply:
add shasum checker